src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -0,0 +1,1325 @@
+                // (9,18): error CS9200: List patterns may not be used for a value of type 'int'.
+                //         _ = 0 is {..0};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForListPattern, "{..0}").WithArguments("int").WithLocation(9, 18),
+                // (9,19): error CS9201: Slice patterns may not be used for a value of type 'int'.
+                //         _ = 0 is {..0};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForSlicePattern, "..0").WithArguments("int").WithLocation(9, 19));
+        }
+
+        [Fact]
+        public void ListPattern_MissingMembers_Constructors()
+        {
+            var source = @"
+using System;
+namespace System
+{
+    public struct Index
+    {
+    }
+    public struct Range
+    {
+    }
+}
+class X
+{
+    public int this[Range i] => 1;
+    public int this[Index i] => 1;
+    public int Length => 1;
+
+    public static void Main()
+    {
+        _ = new X() is { 1 };
+        _ = new X() is { .. 1 };
+    } 
+}
+";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularWithListPatterns, options: TestOptions.ReleaseExe);
+            compilation.VerifyEmitDiagnostics(
+                // (20,24): error CS0656: Missing compiler required member 'System.Index..ctor'
+                //         _ = new X() is { 1 };
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "{ 1 }").WithArguments("System.Index", ".ctor").WithLocation(20, 24),
+                // (21,24): error CS0656: Missing compiler required member 'System.Index..ctor'
+                //         _ = new X() is { .. 1 };
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "{ .. 1 }").WithArguments("System.Index", ".ctor").WithLocation(21, 24),
+                // (21,26): error CS0656: Missing compiler required member 'System.Range..ctor'
+                //         _ = new X() is { .. 1 };
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, ".. 1").WithArguments("System.Range", ".ctor").WithLocation(21, 26));
+        }
+
+        [Fact]
+        public void ListPattern_MissingMembers_ArrayLength()
+        {
+            var source = @"
+class X
+{
+    public static void Main()
+    {
+        _ = new int[0] is [0];
+        _ = new int[0] is {0};
+        _ = new int[0] is {.._};
+    } 
+}
+";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularWithListPatterns, options: TestOptions.ReleaseExe);
+            compilation.MakeMemberMissing(SpecialMember.System_Array__Length);
+            // PROTOTYPE(list-patterns) Nope
+            compilation.VerifyEmitDiagnostics();
+            CompileAndVerify(compilation);
+        }
+
+        [Fact]
+        public void ListPattern_MissingMembers_Substring()
+        {
+            var source = @"
+class X
+{
+    public static void Main()
+    {
+        _ = """" is {.. var slice};
```

Comment:
nit: Consider adding a slice operation `""""[^1]` as well to show it would get the same kind of error.

Reply:
See ListPattern_MissingMembers_GetSubArray_01 and ListPattern_MissingMembers_GetSubArray_02.
In presence of an indexer access we won't get the error on the slice pattern (?!)